runtime/core/portable_type/c10/c10/util/irange.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
```

Comment:
+1

Add BSD license header

Also, i have a task to add a lint rule: https://github.com/pytorch/executorch/issues/8418